botocore/auth.py

```diff
@@ -344,14 +341,16 @@ def _set_necessary_date_headers(self, request):
         # header.  Otherwise we use the X-Amz-Date header.
         if 'Date' in request.headers:
             del request.headers['Date']
+            datetime_timestamp = datetime.datetime.strptime(
+                request.context['timestamp'], '%Y%m%dT%H%M%SZ')
```

Comment:
I suppose it's probably fine either way and is really a very minor point considering out bottleneck is usually the network. Go ahead and keep it if you like, but if you do then I suggest at the very least making the format string a constant, because having it in two places introduces an extra maintenance burden.